From 1c6a744a36ad094c0a6e1b679d889ede8aa715c1 Mon Sep 17 00:00:00 2001
From: Colin Ian King <colin.king@canonical.com>
Date: Thu, 15 Oct 2020 13:22:26 +0100
Subject: [PATCH 15/18] vfio/fsl-mc: fix the return of the uninitialized
 variable ret

Currently the success path in function vfio_fsl_mc_reflck_attach is
returning an uninitialized value in variable ret. Fix this by setting
this to zero to indicate success.

Fixes: f2ba7e8c947b ("vfio/fsl-mc: Added lock support in preparation for interrupt handling")
Signed-off-by: Colin Ian King <colin.king@canonical.com>
Link: https://lore.kernel.org/r/20201015122226.485911-1-colin.king@canonical.com
Addresses-Coverity: ("Uninitialized scalar variable")
Signed-off-by: Nathan Chancellor <natechancellor@gmail.com>
---
 drivers/vfio/fsl-mc/vfio_fsl_mc.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/vfio/fsl-mc/vfio_fsl_mc.c b/drivers/vfio/fsl-mc/vfio_fsl_mc.c
index 80fc7f4ed343..42a5decb78d1 100644
--- a/drivers/vfio/fsl-mc/vfio_fsl_mc.c
+++ b/drivers/vfio/fsl-mc/vfio_fsl_mc.c
@@ -84,6 +84,7 @@ static int vfio_fsl_mc_reflck_attach(struct vfio_fsl_mc_device *vdev)
 		vfio_fsl_mc_reflck_get(cont_vdev->reflck);
 		vdev->reflck = cont_vdev->reflck;
 		vfio_device_put(device);
+		ret = 0;
 	}
 
 unlock:
-- 
2.29.0.rc1

